homeassistant/components/light/lw12wifi.py

```diff
@@ -0,0 +1,210 @@
+
+
+REQUIREMENTS = ['lw12==0.9.2']
+
+_LOGGER = logging.getLogger(__name__)
+
+CONF_TRANSITION = 'transition'
+
+DEFAULT_NAME = 'LW-12 FC'
+DEFAULT_PORT = 5000
+DEFAULT_RGB = [255, 255, 255]
+DEFAULT_BRIGHTNESS = 255
+DEFAULT_EFFECT = None
+DEFAULT_TRANSITION = 128
+
+DOMAIN = 'lw12wifi'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_RGB, default=DEFAULT_RGB):
+    vol.All(list, vol.Length(min=3, max=3),
+            [vol.All(vol.Coerce(int), vol.Range(min=0, max=255))]),
+    vol.Optional(CONF_BRIGHTNESS, default=DEFAULT_BRIGHTNESS):
+    vol.All(vol.Coerce(int), vol.Range(min=0, max=255)),
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_TRANSITION, default=DEFAULT_TRANSITION):
+    vol.All(vol.Coerce(int), vol.Range(min=0, max=255)),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup LW-12 WiFi LED Controller platform."""
+    import lw12
+
+    # Assign configuration variables.
+    name = config.get(CONF_NAME)
+    host = config.get(CONF_HOST)
+    port = config.get(CONF_PORT)
+    rgb = config.get(CONF_RGB)
+    brightness = config.get(CONF_BRIGHTNESS)
+    effect = config.get(CONF_EFFECT)
+    transition = config.get(CONF_TRANSITION)
+    # Add devices
+    add_devices([LW12WiFi(name, host, port, rgb, brightness, effect,
+                          transition)])
+
+
+class LW12WiFi(Light):
+    """LW-12 WiFi LED Controller."""
+
+    def __init__(self, name, host, port, rgb_color, brightness, effect,
+                 transition):
+        import lw12
+
+        self._light = lw12.LW12Controller(host, port)
+        self._name = name
+        self._host = host
+        self._port = port
+        self._rgb_color = rgb_color
+        self._brightness = brightness
+        self._state = None
+        self._effect = effect
+        self._transition_speed = transition
+        # Setup feature list
+        self._supported_features = SUPPORT_BRIGHTNESS
+        self._supported_features |= SUPPORT_EFFECT
+        self._supported_features |= SUPPORT_FLASH
+        self._supported_features |= SUPPORT_RGB_COLOR
+        self._supported_features |= SUPPORT_TRANSITION
+
+    @property
+    def name(self):
+        """Return the display name of the controlled light."""
+        return self._name
+
+    @property
+    def brightness(self):
+        """Return the brightness of the light."""
+        return self._brightness
+
+    @property
+    def rgb_color(self):
+        """Read back the color of the light.
+
+        Returns [r, g, b] list with values in range of 0-255.
+        """
+        return self._rgb_color
+
+    @property
+    def effect(self):
+        """Return current light effect."""
+        try:
+            return self._effect.replace('_', ' ').title()
+        except AttributeError:
+            return None
```

Comment:
Please do the following instead for `None` checks (I'm assuming this is checking for None):

```python
if self._effect is None:
    return None
return self._effect.replace('_', ' ').title()
```

It's just so much easier to understand.